lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2715,6 +2717,46 @@ static T* castWithDebugInfo(SILInstruction *inst) {
   return cast<T>(inst);
 }
 
+/// Given an input type such as $Builtin.Int1, return the type decl of the
+/// corresponding stdlib numeric type, such as $Bool.
+/// See extractBuiltinTypeFromStdlibNumericType() on the reverse type
+/// conversion.
+static NominalTypeDecl *
+getStdlibNumericTypeDeclFromBuiltinType(Type ty, ASTContext &ctx) {
+  // BuiltinIntegerType doesn't carry sign information, which TensorFlow needs,
+  // so we can't rely on getting type information from the builtin types
+  // themselves.  For now we'll just use signed types.
+  if (auto *BII = ty->getAs<BuiltinIntegerType>()) {
+    switch (BII->getFixedWidth()) {
+    case 1:
+      return ctx.getBoolDecl();
+    case 8:
+      return ctx.getInt8Decl();
+    case 16:
+      return ctx.getInt16Decl();
+    case 32:
+      return ctx.getInt32Decl();
+    case 64:
+      return ctx.getInt64Decl();
+    }
+  }
+
+  if (auto *BIF = ty->getAs<BuiltinFloatType>()) {
+    switch (BIF->getFPKind()) {
+    case BuiltinFloatType::IEEE32:
+      return ctx.getFloatDecl();
+    case BuiltinFloatType::IEEE64:
+      return ctx.getDoubleDecl();
+    case BuiltinFloatType::IEEE16:
+    case BuiltinFloatType::IEEE80:
```

Comment:
FYI, the stdlib does have a `Float80` type. It's just that TensorFlow support doesn't need to care about it. Perhaps specify in doc comment that we only care about types are accelerable by tensorflow?

Reply:
Good point. Added to the comment.